src/Core/src/Stream/StreamFactory.php

```diff
@@ -17,9 +17,13 @@ public static function create($content): RequestStream
             return StringStream::create($content ?? '');
         }
         if (\is_callable($content)) {
-            return RewindableStream::create(CallableStream::create($content));
+            return CallableStream::create($content);
         }
         if (\is_iterable($content)) {
+            if ($content instanceof \Generator) {
+                return GeneratorIterableStream::create($content);
```

Comment:
This is a special case for \Generator (read-once) while simple array can be read several time.

I wonder if ArrayObject should be read once too?....

MayBe I should use
```php
if (is_array) 
  return ArrayStream();
else
  return IterableStream();
```

Reply:
Or just add the `ReadOnceResultStream` on `IterableStream`?